app/src/main/java/org/thoughtcrime/securesms/conversation/newmessage/NewMessageFragment.kt

```diff
@@ -0,0 +1,182 @@
+                    onBack = { delegate.onDialogBackPressed() },
+                    onHelp = { requireContext().showOpenUrlDialog("https://sessionapp.zendesk.com/hc/en-us/articles/4439132747033-How-do-Session-ID-usernames-work") }
+                )
+            }
+        }
+    }
+
+    private fun createPrivateChat(hexEncodedPublicKey: String) {
+        val recipient = Recipient.from(requireContext(), Address.fromSerialized(hexEncodedPublicKey), false)
+        Intent(requireContext(), ConversationActivityV2::class.java).apply {
+            putExtra(ConversationActivityV2.ADDRESS, recipient.address)
+            setDataAndType(requireActivity().intent.data, requireActivity().intent.type)
+            putExtra(ConversationActivityV2.THREAD_ID, DatabaseComponent.get(requireContext()).threadDatabase().getThreadIdIfExistsFor(recipient))
+        }.let(requireContext()::startActivity)
+        delegate.onDialogClosePressed()
+    }
+}
+
+@Preview
+@Composable
+private fun PreviewNewMessage(
+    @PreviewParameter(ThemeResPreviewParameterProvider::class) themeResId: Int
+) {
+    PreviewTheme(themeResId) {
+        NewMessage(State())
+    }
+}
+
+private val TITLES = listOf(R.string.enter_account_id, R.string.qrScan)
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun NewMessage(
+    state: State,
+    errors: Flow<String> = emptyFlow(),
+    callbacks: Callbacks = object: Callbacks {},
+    onClose: () -> Unit = {},
+    onBack: () -> Unit = {},
+    onHelp: () -> Unit = {},
+) {
+    val pagerState = rememberPagerState { TITLES.size }
+
+    Column(modifier = Modifier.background(MaterialTheme.colors.primarySurface)) {
+        AppBar(stringResource(R.string.messageNew), onClose = onClose, onBack = onBack)
+        SessionTabRow(pagerState, TITLES)
+        HorizontalPager(pagerState) {
+            when (TITLES[it]) {
+                R.string.enter_account_id -> EnterAccountId(state, callbacks, onHelp)
+                R.string.qrScan -> MaybeScanQrCode(errors, onScan = callbacks::onScanQrCode)
+            }
+        }
+    }
+}
+
+@Composable
+fun EnterAccountId(
+    state: State,
+    callbacks: Callbacks,
+    onHelp: () -> Unit = {}
+) {
+    Column(
+        modifier = Modifier.padding(horizontal = 12.dp, vertical = 16.dp).fillMaxHeight(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(16.dp)
+    ) {
+        SessionOutlinedTextField(
+            text = state.newMessageIdOrOns,
+            modifier = Modifier.padding(horizontal = 64.dp)
+                .contentDescription("Session id input box"),
+            placeholder = stringResource(R.string.accountIdOrOnsEnter),
+            onChange = callbacks::onChange,
+            onContinue = callbacks::onContinue,
+            error = state.error?.string(),
+        )
+        if (state.error == null) {
+            BorderlessButtonSecondary(
```

Comment:
If you drill down into this composable it uses extraSmall, though Figma has it set to `small`. At least here, no sure if this applies for everything you have set to BorderlessButtonSecondary. This one needs to change but probably all of them. If so you can change the text style down into this component.